pkg/controller/spire-oidc-discovery-provider/configmaps.go

```diff
@@ -4,31 +4,30 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"strings"
+
 	"github.com/openshift/zero-trust-workload-identity-manager/api/v1alpha1"
 	"github.com/openshift/zero-trust-workload-identity-manager/pkg/controller/utils"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // GenerateOIDCConfigMapFromCR creates a ConfigMap for the spire oidc discovery provider from the CR spec
-func GenerateOIDCConfigMapFromCR(cr *v1alpha1.SpireOIDCDiscoveryProvider) (*corev1.ConfigMap, error) {
-	if cr == nil {
+func GenerateOIDCConfigMapFromCR(dp *v1alpha1.SpireOIDCDiscoveryProvider) (*corev1.ConfigMap, error) {
+	if dp == nil {
 		return nil, errors.New("spire OIDC Discovery Provider Config is nil")
 	}
 	// Default to "spire-agent.sock" if not provided
-	agentSocketName := cr.Spec.AgentSocketName
+	agentSocketName := dp.Spec.AgentSocketName
 	if agentSocketName == "" {
 		agentSocketName = "spire-agent.sock"
 	}
 
 	// Determine trust domain
-	trustDomain := cr.Spec.TrustDomain
+	trustDomain := dp.Spec.TrustDomain
 
 	// JWT Issuer fallback
-	jwtIssuer := cr.Spec.JwtIssuer
-	if jwtIssuer == "" {
-		jwtIssuer = fmt.Sprintf("oidc-discovery.%s", trustDomain)
-	}
+	jwtIssuer := stripProtocol(dp.Spec.JwtIssuer)
```

Comment:
The dp.Spec.JwtIssuer must be parsed for validating the URL to fail on un-intended formats.
Ref: https://github.com/spiffe/spire/blob/363a1074f12d4ef67a046d09682e95e6ae370658/support/oidc-discovery-provider/main.go#L74 
The jwt issuer configuration is used to build oidc discovery document which will be referred by applications to obtain specific endpoint URL paths for further validation process.
Ref: https://github.com/spiffe/spire/blob/363a1074f12d4ef67a046d09682e95e6ae370658/support/oidc-discovery-provider/handler.go#L66